cpp-linter/src/clang_tools/clang_tidy.rs

```diff
@@ -281,10 +277,12 @@
     let mut original_content = None;
     if clang_params.tidy_review {
         cmd.arg("--fix-errors");
-        original_content =
-            Some(fs::read_to_string(&file.name).expect(
-                "Failed to cache file's original content before applying clang-tidy changes.",
-            ));
+        original_content = Some(fs::read_to_string(&file.name).with_context(|| {
+            format!(
+                "Failed to cache file's original content before applying clang-tidy changes: {}",
+                file_name.clone()
+            )
+        })?);
```

Comment:
I don't think this is an "easy" request. It would require the test to delete the file while the function is running, and there's no way to guarantee the test outcome at that point.

> [!note]
> The file analyzed by this function is guaranteed to exist before calling it.

If this error occurs, then it would be pretty rare. And having more details about how to reproduce it would provide an avenue to creating reproducible test results. Giving the error a detailed context should be enough for users to easily avoid this.